packages/api-graphql/__tests__/GraphQLAPI-test.ts

```
@@ -755,7 +755,7 @@ describe('API test', () => {
 					variables,
 					authMode: GRAPHQL_AUTH_MODE.OPENID_CONNECT,
 				})
-			).rejects.toThrowError('No federated jwt');
+			).rejects.toThrowError('No current user');
```

The previous test case intents to test when able to getItem but no `token` prop in [this line](https://github.com/aws-amplify/amplify-js/blob/36de9014320f2a39caba492ad156da4f6a79575a/packages/api-graphql/src/GraphQLAPI.ts#L171)

So we may not want to change the expected result here but change the mock to be correctly mocked instead. Perhaps, change the `jest.spyOn(Cache, 'getItem').mockReturnValue(null);` to `jest.spyOn(Cache, 'getItem').mockReturnValue({});` should make it correctly tested.

---

Fixed. I need to make some changes to `GraphQLAPI.ts` as well because the try/catch was swallowing the `No federated jwt` error.